### PR TITLE
feat: slidespeak slides generation

### PR DESCRIPTION
**Pull request by fdb02983rhy**
# Summary

image

# Checklist

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

### CODE REVIEW

**Review by crazywoola — APPROVED** [review]

File: api/core/tools/provider/builtin/slidespeak/slidespeak.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        if base_url:
+            base_url = str(URL(base_url) / "v1")
+
+        headers = {"Content-Type": "application/json", "X-API-Key": api_key}
+
+        test_task_id = "xxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+        url = f"{base_url or 'https://api.slidespeak.co/api/v1'}/task_status/{test_task_id}"
+
+        response = requests.get(url, headers=headers)
+        if response.status_code != 200:
```
Comment (kgoedecke) [inline api/core/tools/provider/builtin/slidespeak/slidespeak.py:27]: I think this check is invalid...

Comment (fdb02983rhy) [reply api/core/tools/provider/builtin/slidespeak/slidespeak.py:27]: > I think this check is invalid...

I tested this approach before— is it not working now?
FYI, tools will function as plugins in the upcoming release. If a fix is needed, it’ll be merged after the release.

Comment (kgoedecke) [reply api/core/tools/provider/builtin/slidespeak/slidespeak.py:27]: @fdb02983rhy We will pick this up from our end later today and open a PR. When is the release scheduled?

Comment (fdb02983rhy) [reply api/core/tools/provider/builtin/slidespeak/slidespeak.py:27]: @crazywoola FYI

Comment (fdb02983rhy) [reply api/core/tools/provider/builtin/slidespeak/slidespeak.py:27]: > @fdb02983rhy We will pick this up from our end later today and open a PR. When is the release scheduled?

Oh, I just realized you’re from SlideSpeak—thanks for creating such a great SaaS! I actually tried introducing it to my company’s management a while back, but at the time, it couldn’t generate slides directly from user-inputted text (not just from a topic or title, but from the actual text provided). I’d love it if your team could consider adding this feature to the API.

File: api/core/tools/provider/builtin/slidespeak/slidespeak.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        if not api_key:
+            raise ToolProviderCredentialValidationError("API key is missing")
+
+        if base_url:
+            base_url = str(URL(base_url) / "v1")
+
+        headers = {"Content-Type": "application/json", "X-API-Key": api_key}
+
+        test_task_id = "xxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+        url = f"{base_url or 'https://api.slidespeak.co/api/v1'}/task_status/{test_task_id}"
```
Comment (pulgamecanica) [inline api/core/tools/provider/builtin/slidespeak/slidespeak.py:24]: You should use `https://api.slidespeak.co/api/v1/me`

File: api/core/tools/provider/builtin/slidespeak/slidespeak.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+        response = requests.get(url, headers=headers)
+        if response.status_code != 200:
+            raise ToolProviderCredentialValidationError("Invalid SlidePeak API key")
```
Comment (pulgamecanica) [inline api/core/tools/provider/builtin/slidespeak/slidespeak.py:28]: Typo, should be:

```py

raise ToolProviderCredentialValidationError("Invalid SlideSpeak API key")
```

**Pull request merged**